src/rockstor/fs/btrfs.py

```diff
@@ -85,46 +85,110 @@ def add_pool(pool, disks):
     return out, err, rc
 
 
+def is_pool_missing_dev(label):
+    """
+    Simple and fast wrapper around 'btrfs fi show --raw label' to return True /
+    False depending on if a device is reported missing from the given pool by
+    label. Works by matching the end of output lines for the string 'missing',
+    which is either, post remount reboot (first and last populated line) or
+    the last line (dev removed and no remount).
+    Label is used as this is preserved in our Pool db so will work if the pool
+    fails to mount, and there by allows surfacing this as a potential reason
+    for the mount failure.
+    :param label: Pool label.
+    :return: True if at least one device was found to be missing, False if not.
+    """
+    if label is None:
+        return False
+    # --raw used to minimise pre-processing of irrelevant 'used' info (units).
+    cmd = [BTRFS, 'fi', 'show', '--raw', label]
+    o, e, rc = run_command(cmd)
+    if o[-3].endswith('missing') or o[0].endswith('missing'):
```

Comment:
this seems fuzzy. If btrfs is inconsistent, may be better to iterete over all output lines and search for the matching substring. Also, can the substring be a bit more specific than just the word "missing"?

Reply:
"... may be better to iterete over all output lines and search for the matching substring."
My concern here was to be as light/simple and fast as possible as we are a model property at this point.

I know what you mean though as I had originally just searched all lines for string.endswith('missing') but took it down to last (non empty) line and doubled checked against the first line just in case as this was the inconsistent one. We can later move to libbtrfsutil for this kind of stuff, once btrfs-progs is rebased on it and it becomes canonical. Should help to remove all this less than ideal parsing. Prior to that we can use the soon to be added json output switch which will be nice.

As for "missing" that seemed to cover all that I saw which was really only:
"*** Some devices missing" on last non empty line.
and the likes of:
"warning, device 5 is missing" which obviously varied (by id) and was also sometime repeated verbatim !!
Hence "missing" seemed to cover it.

Looking forward to json output options and the new libtrfsutil :)